### ISSUE TITLE
crash when login twice

### ISSUE THREAD

**Issue by btcid** [blockchain/service-my-wallet-v3#17]
Hi,

When my app try to login but it's already logged in, it crashes:

```
events.js:142
      throw er; // Unhandled 'error' event
      ^

Error: socket hang up
    at createHangUpError (_http_client.js:210:15)
    at TLSSocket.socketOnEnd (_http_client.js:295:23)
    at emitNone (events.js:73:20)
    at TLSSocket.emit (events.js:167:7)
    at endReadableNT (_stream_readable.js:906:12)
    at nextTickCallbackWith2Args (node.js:455:9)
    at process._tickCallback (node.js:369:17)
```

Is this expected behaviour? Is there any method to check if we already logged in, to prevent calling the login request again.

Thanks

**Comment by jtormey** [comment]
Hi @btcid. A HangUpError is emitted when a request goes out and the server never sends back a response. This might be caused by sending many requests to the /login endpoint in a short amount of time.

Regardless, this should not be allowed to crash the program. I'll create a fix and close this issue when that's done. Thanks for letting me know about this problem.

**Comment by jtormey** [comment]
Fixed in v0.10.0, commit d2ac7fa6
